### ISSUE TITLE
Doesn't Work

### ISSUE THREAD

**Issue by pavlibeis** [tonyd256/TDAudioStreamer#1]
I manage to connect the 2 devices and when i select a song i see the album picture in both. But I hear no music, no sound, nothing. Are you sure its working ?? Tested with  iphone + ipad, iphone + simulator, with no success.

**Comment by ghost** [comment]
me too ...

**Comment by jamesanse** [comment]
i managed to get it to work, make sure both are running on device with iOS7

**Comment by pavlibeis** [comment]
Both my iphone and Ipad (but also the simulator) have ios7. Really don't understand why it doesn't work.

**Comment by veeralpatel** [comment]
Yep, it's not working.

**Comment by pavlibeis** [comment]
Hey guys, found the problem. If you have a song with an album picture on it then it won't work. But if the song has no picture, it works fine.

**Comment by augmentedminds** [comment]
I don't believe that it is an album picture issue, but an issue with playing anything other then an .mp3 (like an m4a - which usually has album art). I will open a new ticket with this.

**Comment by karthikprabhuA** [comment]
me too.. .mp3 is not working

**Comment by spenciefy** [comment]
I sorta got it to work by putting [self start]; in TDAudioOutputStreamer after line 101... but it only plays the music on the guest device. How can I play the music simultaneously from the host device? Also, I can't seem to get 3+ phones connected to play...

**Comment by alariccole** [comment]
I've confirmed it working with an mp3, but not m4a. No ideas as to why, either. @spenciefy does that [self start] call still work. Just after the for... loop?

**Comment by spenciefy** [comment]
Yeah, I realized that it only works with mp3. I think the way the audio files are streamed don't work with m4a. I don't think the [self start] worked, but you could try.

**Comment by eduard23D** [comment]
Hi, Everything works great for me with a mp3, but I have a problem when i have 2 devices like guest (2 iphones or (1 iphone and 1 simulator)) and 1 device(simulator) like a host. The song is playing just in one guest. Someone have same problem ? Thanks . 

UPDATE: Solved my problem, In TDMultipeerHostViewController.m  file, at line 70, the method [ ... initWithOutputStream:[self.session outputStreamForPeer:peers[0]]]; is just for first peer connected.

**Comment by adamlangsner** [comment]
The majority of my library is mp3s and some songs work while some don't. I've noticed that the ones that don't play have a bitrate higher than 128kbps. Not sure where to go from there though. Is there a way I could compress my higher bitrate songs on the fly?

**Comment by jamesanse** [comment]
according to this doc https://developer.apple.com/library/ios/documentation/musicaudio/reference/AudioStreamReference/Reference/reference.html

Audio File Stream Services supports the following audio data types:

AIFF
AIFC
WAVE
CAF
NeXT
ADTS
MPEG Audio Layer 3
AAC

So m4a is not supported

**Comment by ghost** [comment]
Did you try with some AAC Audio files?, because doesn't work too..... and with some AIFF files?  i guess should exist anyway for m4a im searching, but still without luck.

If you change the type of  the hint in AudioFileStreamOpen to kAudioFileAAC_ADTSType, not get more unsupported file error, but fails when try to fill the buffer.  EXEC_BAD_ACCESS

**Comment by karthikprabhuA** [comment]
no

On Tue, May 6, 2014 at 4:10 AM, Gonzalo David Bahamondez Coffre <
notifications@github.com> wrote:

> did you try with some AAC Audio files?
> 
> —
> Reply to this email directly or view it on GitHubhttps://github.com/tonyd256/TDAudioStreamer/issues/1#issuecomment-42249412
> .

## 

with regards,
Karthik Prabhu.A

**Comment by ghost** [comment]
This works ,  only with mp3 file :/ , should work , with  all Audio file stream services supported formats, i think something is wrong.

**Comment by nyanev** [comment]
I think the problem is with DRM of songs. If you bought some music, you can't stream it.

**Comment by sound-eng** [comment]
encoded mp3 file as m4a (obviously without DRM), put it into itunes lib on my ipad, iTunes can play it, while TDAudioStreamer can't. Probably the problem is in file format header which AVAsserReader does not pass through - only media data.

**Comment by kristersj** [comment]
@tonyd256 How can I change the outputStreamForPeer method to stream to multiple peers instead of just one?

**Comment by primedev-iOS** [comment]
How to use outputStreamForPeer for multiple connected peers and how to play the same audio for host also?

**Comment by primedev-iOS** [comment]
Whenever I am using loop for this above task there is some gap between all devices.
Please help.

**Comment by primedev-iOS** [comment]
How to use outputStreamForPeer for multiple connected peers and how to play the same audio for host also?
Anybody wants to share his/her knowledge here or humanity is dead.
